src/libraries/System.Private.CoreLib/src/System/Reflection/TypeDelegator.cs

```diff
@@ -20,11 +20,12 @@ public override bool IsAssignableFrom([NotNullWhen(true)] TypeInfo? typeInfo)
             return IsAssignableFrom(typeInfo.AsType());
         }
 
+        [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)]
         protected Type typeImpl = null!;
 
         protected TypeDelegator() { }
 
-        public TypeDelegator(Type delegatingType)
+        public TypeDelegator([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)] Type delegatingType)
```

Comment:
Linker treats this constructor as an intrinsic that just propagates whatever metadata we have about `delegatingType` to the newly allocated object. We need this annotation so that the analysis of methods touching `typeImpl` don't warn about unsafe reflection use.

Reply:
We should put that as a comment into the code.

Reply:
Done.